### PR TITLE
Revise the translation for Korean

### PR DESCRIPTION
**Pull request by jiyongp**
- Run msgmerge --no-wrap --update po/ko.po po/messages.pot
- Be consistent with the line ending. For example, if the original text ends with ":", so the translation is.
- Remove unnecessary newline characters
- Fix the remaining inconsistencies on line ending chracters
- s/제너릭/제네릭/g
- s/트레이트/트레잇/g

### CODE REVIEW

File: po/ko.po [hunk 1]
```diff
@@ -1168,60 +1132,6 @@ msgstr ""
 "* <kbd>Ctrl + Enter</kbd>: 현재 포커스를 받은 코드 샘플 블록을 실행합니다.\n"
 "* <kbd>s</kbd>: 검색창을 활성화합니다.(mdbook 문제로 23.01.19 기준 영어로만 가능합니다.)"
 
-#: src/running-the-course/translations.md:1
```
Comment (mgeisler) [inline po/ko.po:1171]: It looks like this got deleted by accident?

File: po/ko.po [hunk 2]
```diff
@@ -29,755 +29,751 @@ msgid "Keyboard Shortcuts"
 msgstr "단축키"
 
 #: src/SUMMARY.md:7
-msgid "Translations"
-msgstr "다른 언어들"
```
Comment (jooyunghan) [inline po/ko.po:33]: Is this missing?

Comment (jooyunghan) [reply po/ko.po:33]: You probably need to generate po/messages.pot from the HEAD first.

**Review by jooyunghan — COMMENTED** [review]
+1

Btw, can we keep hard line-breaking for better readability in source view?

**Comment by jooyunghan** [comment]
Beside it will help when editing long bullet items.

**Review by jooyunghan — DISMISSED** [review]

File: po/ko.po (hunk 1 above)
Comment (jiyongp) [reply po/ko.po]: Sorry, and thanks for pointing that out. Fixed.

File: po/ko.po (hunk 2 above)
Comment (jiyongp) [reply po/ko.po]: Done

**Comment by jiyongp** [comment]
> Btw, can we keep hard line-breaking for better readability in source view?

I think it could/should be done in the editor you are using, until when we have a consensus on the text width in the po files.

**Review by jooyunghan — APPROVED** [review]
Unfortunately, editors (at least POEdit for now) don't handle line-wrapping well with non-english texts. It's because, byte length and visual font width for a character don't match. For readability, line-wrapping should be based on rendered width of texts but most editors do that based on bytes. For example, 가 (Korean) has 3-byte length (in utf8) while it's typically rendered as wide as two latin characters in most of fixed fonts.

**Comment by jiyongp** [comment]
I meant that you can resize your editor's width (not the textwidth setting which the editor uses when writing the contents to the file) if a long text is hard to read & edit.

**Comment by jooyunghan** [comment]
> Beside it will help when editing long bullet items.

Even when we set the editor's width, it will look like
```
* 무궁화 꽃이 피었습니다. 무궁화 꽃이 피었습니다. 무궁화 
꽃이 피었습니다. 무궁화 꽃이 피었습니다.
  * 무궁화 꽃이 피었습니다. 무궁화 꽃이 피었습니다. 무궁
화 꽃이 피었습니다. 꽃이 피었습니다. 무궁화 꽃이 피었습
니다.
  * 무궁화 꽃이 피었습니다. 무궁화 꽃이 피었습니다. 무궁
화 꽃이 피었습니다. 꽃이 피었습니다. 무궁화 꽃이 피었습
니다.
```
while I hope ..
```
* 무궁화 꽃이 피었습니다. 무궁화 꽃이 피었습니다. 무궁화 
  꽃이 피었습니다. 무궁화 꽃이 피었습니다.
  * 무궁화 꽃이 피었습니다. 무궁화 꽃이 피었습니다. 무궁
    화 꽃이 피었습니다. 꽃이 피었습니다. 무궁화 꽃이 피
    었습니다.
  * 무궁화 꽃이 피었습니다. 무궁화 꽃이 피었습니다. 무궁
    화 꽃이 피었습니다. 꽃이 피었습니다. 무궁화 꽃이 피
    었습니다.
```

**Comment by jiyongp** [comment]
We can't do that as long as we use poedit. It forcibly adds a new line character. For example, 

What we type in poedit:
```
정상적으로 설치가 되었으면 강의의 코드 블록중 하나를
아래 단계를 따라 로컬에서 실행할 수 있습니다:
```

What poedit saves to the po file:

```
msgstr ""
"정상적으로 설치가 되었으면 강의의 코드 블록중 하나를\n"
"아래 단계를 따라 로컬에서 실행할 수 있습니다:"
```

I wish it were:
```
msgstr ""
"정상적으로 설치가 되었으면 강의의 코드 블록중 하나를 "
"아래 단계를 따라 로컬에서 실행할 수 있습니다:"
```

I think the problem with bullets should be addressed by #318.

**Comment by jooyunghan** [comment]
I liked 
```
msgstr ""
"정상적으로 설치가 되었으면 강의의 코드 블록중 하나를\n"
"아래 단계를 따라 로컬에서 실행할 수 있습니다:"
```
because at least it gives me the control. It keeps `\n` exactly where I typed `<cr>`.

But, yeah, it depends on personal preference. If you don't like it, it's okay. It's already merged ;-)

**Comment by jiyongp** [comment]
That newline character in the translation affects the rendering, which I think is bad.

```
msgstr ""
"이 4일짜리 러스트 강의는 안\n"
"드로이드 팀이 만들었습니다.
```

is rendered as 

```
이 4일짜리 러스트 강의는 안 드로이드 팀이 만들었습니다. 
```

Notice the unexpected space after "안".

**Comment by jiyongp** [comment]
In addition, the generated html is as follows:

```
<p>이 4일짜리 러스트 강의는 안
드로이드 팀이 만들었습니다.</p>
```

Nobody will read the html, but I think that the translation shouldn't affect any aspect of the output formatting.

**Comment by jooyunghan** [comment]
Oh, I see. You're right. That's also the problem I wanted to avoid (unexpected new lines, hence affecting the output formatting). I described how I avoided it in the commit 6aa1aac16580d4442034da61234677efbdb55a96.

Poedit's settings:
- uncheck "line-wrapping"
- uncheck "keep old formatting"

By these settings (along with --no-wrap), newlines are all under control(no more automatic/unexpected line-breakings).

**Comment by mgeisler** [comment]
> That newline character in the translation affects the rendering, which I think is bad.
> 
> ```
> msgstr ""
> "이 4일짜리 러스트 강의는 안\n"
> "드로이드 팀이 만들었습니다.
> ```
> 
> is rendered as
> 
> ```
> 이 4일짜리 러스트 강의는 안 드로이드 팀이 만들었습니다. 
> ```
> 
> Notice the unexpected space after "안".

Yeah, that's unfortunate!

You're editing Markdown, so regardless of the formatting of the `.po` file, it's the rules of Markdown that matter for the final formatting.

I think I see some spaces im the text above, so could you not break the line after those?

**Comment by jiyongp** [comment]
> could you not break the line after those?

Yes, this is what I did in the commit 6841827

**Pull request merged**